### ISSUE TITLE
Cannot edit Postgres citext cells

### ISSUE THREAD

**Issue by adminsyszg** [dbeaver/dbeaver#17108]
#### System information: 
- Operating system: **Windows 7 pro**
- DBeaver version:  **22.1.2.202207091909**

#### Connection specification:
- Database name and version: **Postgresql 10.4**
- Driver name: **Postgresql** 
- Do you use tunnels or proxies (SSH, SOCKS, etc)? **No**

#### Describe the problem you're observing:

When i try to edit cell from a View Data witch datatype Citext , DBeaver prevents the edit with a Column "ColumnName" is read-only: No corresponding table column message

![изображение](https://user-images.githubusercontent.com/70959040/179453176-d7a09660-e9a8-4fd4-b264-49c9e298620e.png)

![изображение](https://user-images.githubusercontent.com/70959040/179453221-12d34a0c-e2d2-42ba-bd93-c22075a26d91.png)

DDL of table
```
`-- apteka.salestableapreserve definition

-- Drop table

-- DROP TABLE apteka.salestableapreserve;

CREATE TABLE apteka.salestableapreserve (
recid serial4 NOT NULL,
dataareaid public.citext NULL,
tableid int4 NULL DEFAULT 50558,
createddate timestamp NULL DEFAULT 'now'::text::timestamp without time zone,
modifieddate timestamp NULL DEFAULT 'now'::text::timestamp without time zone,
modifiedby public.citext NULL,
createdby public.citext NULL DEFAULT "current_user"(),
origrecid int4 NULL,
custaccount public.citext NOT NULL DEFAULT ''::citext,
custname public.citext NOT NULL DEFAULT ''::citext,
custphone public.citext NULL DEFAULT ''::citext,
delivery bool NOT NULL DEFAULT false,
deliverydate apteka.transdate NOT NULL DEFAULT 'now'::text::timestamp without time zone,
dis public.citext NOT NULL DEFAULT ''::citext,
disccardid apteka.disccardnum NOT NULL DEFAULT ''::citext,
finishdate apteka.transdate NOT NULL DEFAULT 'now'::text::timestamp without time zone,
newdisccard bool NOT NULL DEFAULT false,
"percent" apteka."percent" NOT NULL DEFAULT 0,
reserveby public.citext NOT NULL DEFAULT ''::citext,
reservedate apteka.transdate NOT NULL DEFAULT 'now'::text::timestamp without time zone,
reservedateperiod timestamp NOT NULL DEFAULT 'now'::text::timestamp without time zone,
reservedeliveryaddrid apteka.reservedeliveryaddrid NOT NULL DEFAULT ''::citext,
returnreason apteka.returnreasonid NOT NULL DEFAULT ''::citext,
salesreserveid public.citext NOT NULL DEFAULT ''::citext,
salesreserverefid public.citext NOT NULL DEFAULT ''::citext,
salestype public.citext NOT NULL DEFAULT ''::citext,
status public.citext NOT NULL DEFAULT ''::citext,
street public.citext NOT NULL DEFAULT ''::citext,
doorway public.citext NULL,
deliverytime apteka.reservedeliverytimeid NOT NULL DEFAULT ''::citext,
hlcard public.citext NOT NULL DEFAULT ''::citext,
kkmnum public.citext NOT NULL DEFAULT ''::citext,
inuse bool NOT NULL DEFAULT false,
built bool NOT NULL DEFAULT false,
beznal bool NOT NULL DEFAULT false,
canceled bool NOT NULL DEFAULT false,
reservetype int4 NOT NULL DEFAULT 1,
prepaid bool NOT NULL DEFAULT false,
siteid public.citext NULL,
CONSTRAINT salestableapreserve_recid PRIMARY KEY (recid),
CONSTRAINT tableid CHECK ((tableid = 50558))
);
CREATE UNIQUE INDEX salestableapreserve_salesreserveid_idx ON apteka.salestableapreserve USING btree (salesreserveid, status);

-- Table Triggers

create trigger salestableapreserve_geninsert before
insert
on
apteka.salestableapreserve for each row execute procedure salestableapreserve_numseq();
create trigger salestableapreserve_modifieddate before
update
on
apteka.salestableapreserve for each row execute procedure modifieddate();
create trigger salestableapreserve_update before
update
on
apteka.salestableapreserve for each row execute procedure salestableapreserve_update();`
```

**Comment by LonwoLonwo** [comment]
Thanks for the bug report @adminsyszg 

This issue related to #16733

**Comment by E1izabeth** [comment]
I'm unable to reproduce with 22.1.3. Is this issue actual for the latest DBeaver version?

**Comment by adminsyszg** [comment]
> I'm unable to reproduce with 22.1.3. Is this issue actual for the latest DBeaver version?

No, it's not actual for latest version, work fine. Thx!

**Comment by groogiam** [comment]
@adminsyszg @E1izabeth This is still happening for me on 22.1.3.
![image](https://user-images.githubusercontent.com/1992395/182744392-7ec3096f-f0c1-4236-8c1d-22ed0e43c7eb.png)

PostgreSQL 12.11 on x86_64-pc-linux-musl, compiled by gcc (Alpine 11.2.1_git20220219) 11.2.1 20220219, 64-bit

**Comment by E1izabeth** [comment]
@groogiam please, provide table ddl and steps to reproduce.

**Comment by adminsyszg** [comment]
> @groogiam please, provide table ddl and steps to reproduce.

in some cases, it's still reproduce. in different instaneses DB it work fine or not work. DB-shema - analogue.

**Comment by groogiam** [comment]
@adminsyszg @E1izabeth 

To Reproduce:

1. Open Database Navigator
2. Expand Postgres instance
3. Expand Databases -> Database Name -> Schemas -> public -> Tables
4. Double Click on a table that contains a citext column.
5. On the "Data" tab of the opened window select a row and column. Double click the column to edit.

Expected Result:
The grid column goes into edit mode allow the text to be updated.

Actual Result:

The column stays in read only mode and does not allow any text to be entered. The following status message is shown at the bottom right of the window.

![image](https://user-images.githubusercontent.com/1992395/182744392-7ec3096f-f0c1-4236-8c1d-22ed0e43c7eb.png)

This happens on all databases with citext columns on this instance.

The postgres instance is running in docker using the official PostgreSQL docker images.

https://hub.docker.com/layers/postgres/library/postgres/12-alpine/images/sha256-00481bdc69f21e960c543a19f4406c48c1a559d6af150ba6a099fc93891d8918?context=explore

PostgreSQL 12.11 on x86_64-pc-linux-musl, compiled by gcc (Alpine 11.2.1_git20220219) 11.2.1 20220219, 64-bit

**Comment by LonwoLonwo** [comment]
Hello @groogiam 
Can you provide your table DDL, please?

**Comment by groogiam** [comment]
@LonwoLonwo 

See below ddl for which reproduces the issue. As I've stated in my previous comments it is reproducible for me on all citext columns. Thanks.

CREATE TABLE public.citexteditrepro (
	id uuid NOT NULL,
	loginname public.citext NOT NULL,
	CONSTRAINT pk_citexteditrepro PRIMARY KEY (id)
);

**Comment by E1izabeth** [comment]
Is this issue reproducible with the latest DBeaver version (22.1.4)? Please, try upgrading to the latest DBeaver 22.1.4.

**Comment by groogiam** [comment]
Yes it is still happening on 22.1.4. Thanks.

**Comment by groogiam** [comment]
Is there any update on this? 22.3 has been released and this is still an issue.

**Comment by groogiam** [comment]
Any update on a fix for this?

**Comment by adminsyszg** [comment]
> Any update on a fix for this?

I still have the same problem with dbeaver 23.0.2

**Comment by adminsyszg** [comment]
> Is this issue reproducible with the latest DBeaver version (22.1.4)? Please, try upgrading to the latest DBeaver 22.1.4.

@E1izabeth  
dbeaver 23.0.2 has this problem yet

**Comment by groogiam** [comment]
This appears to be fixed as of 23.3.3.202401211839

**Comment by uslss** [comment]
thanks for update
